adapters/cisco_nx_rest/me_restore_configuration.php

```diff
@@ -62,39 +65,62 @@ function generate_from_old_revision($revision_id)
   }
 
 
-  function restore_conf()
-  {
-    global $apply_errors;
-
-    global $sms_sd_ctx;
-    $ret = SMS_OK;
-
-    $file_name = "{$this->sdid}.cfg";
-    $full_name = $_SERVER['TFTP_BASE'] . "/" . $file_name;
-
-    $ret = save_file($this->runningconf_to_restore, $full_name);
-    if ($ret !== SMS_OK)
-    {
-      return $ret;
-    }
-    $ret = save_result_file($this->runningconf_to_restore, 'conf.applied');
-    if ($ret !== SMS_OK)
-    {
-      return $ret;
-    }
-    
-    $http_header_str = "Content-Type: application/yang.data+xml|Accept: application/yang.data+xml";
-    $sms_sd_ctx->http_header_list = explode("|", $http_header_str);
-    $cmd = "PATCH#/restconf/data/Cisco-NX-OS-device#@$full_name";
-    $ret = $sms_sd_ctx->send(__FILE__ . ':' . __LINE__, $cmd);
-
-    if ($ret !== SMS_OK)
-    {
-      return $ret;
-    }
-    echo "restore_conf method is finished";
-    return $ret;
-  }
+  function restore_conf() {
+  		global $apply_errors;
+
+  		global $sms_sd_ctx;
+  		$ret = SMS_OK;
+
+  		echo "SCP mode configuration\n";
+
+  		// Request flash space on router
+  		$file_name = "{$this->sdid}.cfg";
+  		$full_name = $_SERVER ['TFTP_BASE'] . "/" . $file_name;
+
+  		$ret = save_file ( $this->runningconf_to_restore, $full_name );
+  		if ($ret !== SMS_OK) {
+  			return $ret;
+  		}
+  		$ret = save_result_file ( $this->runningconf_to_restore, 'conf.applied' );
+  		if ($ret !== SMS_OK) {
+  			return $ret;
+  		}
+  		try {
+  			$ret = scp_to_router ( $full_name, $file_name );
+  			if ($ret === SMS_OK) {
+  				// SCP OK
+  				$SMS_OUTPUT_BUF = copy_to_running ( "rollback running-config file bootflash:$file_name" );
```

Comment:
From y point of view , it's better to not specify bootflash: as it's working fine without setting it. For instance, it may hppened for so other Cisco NX device model that it's maybe not bootflash, and in that case this command will failed.